drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
+		crypt = &netlink.XfrmStateAlgo{Name: "rfc4106(gcm(aes))", Key: append(k.value, salt...), ICVLen: 64}
```

Comment:
My problem is: if the key is the same for everyone, we have to guarantee that the salts are different for every single pair of hosts creating an IPSec connection, or else they will eventually collide on the same IV.